src/xmipp/libraries/reconstruction/subtract_projection.cpp

```diff
@@ -662,10 +816,35 @@ void ProgSubtractProjection::processImage(const FileName &fnImg, const FileName
 	I.write(fnImgOut.substr(0, dotPos) + "_I" + fnImgOut.substr(dotPos));
 	#endif
 
-	// Compute particle stats after subtraction
-	double avg;
-	double std;
-	double zScore;
+	// Estimate noise after sutraction
+	if(noiseEstimationBool)
+	{
+		noiseEstimation();
+	}
+
+	writeParticle(rowOut, fnImgOut, Idiff, R2adj(0), beta0save, beta1save, b); 
+}
+
+void ProgSubtractProjection::finishProcessing()
+{
+	if (allow_time_bar && verbose && !single_image)
+        progress_bar(time_bar_size);
+    writeOutput();
+
+	if(noiseEstimationBool)
```

Comment:
I think here you are missing a MPI_Reduce of the noise spectras and n of all nodes